drivers/serial/uart_stm32.c

```diff
@@ -344,6 +332,85 @@ static int uart_stm32_init(struct device *dev)
 	return 0;
 }
 
+
+#ifdef CONFIG_UART_LINE_CTRL
+static int uart_stm32_line_ctrl_set(struct device *dev,
+				    u32_t ctrl, u32_t val)
+{
+	const struct uart_stm32_config *config = DEV_CFG(dev);
+	int result = 0;
+
+#define change_if_not_same(cfg_name, _val)		\
+	do {						\
+		if (config->cfg_name == _val) {		\
+			goto done;			\
+		} else {				\
+			config->cfg_name = _val;	\
+		}					\
+	} while (0)
+
+	switch (ctrl) {
+	case LINE_CTRL_BAUD_RATE:
```

Comment:
You should not store information in baud_rate which is used to hold start-up configuration.
I propose you update registers directly.